src/modules/shortcut_guide/shortcut_guide.cpp

```diff
@@ -52,6 +55,20 @@ namespace
         {
             return result;
         }
+
+        DWORD pid = 0;
+        WCHAR exePath[MAX_PATH]{};
+        GetWindowThreadProcessId(active_window, &pid);
+        if (pid != 0)
+        {
+            HANDLE processHandle = OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, pid);
+            if (processHandle && GetProcessImageFileName(processHandle, exePath, MAX_PATH) > 0)
+            {
+                result.disabled = instance->is_disabled_app(exePath);

```

Comment:
We need some code refactoring, since once you find an app for which the SG is disable, there is no reason to continue processing this request.

Also consider moving the new code to a helper function that as first thing checks if the list of apps is empty and in case returns without trying to retrieve the exe path.

Reply:
I have converted the PR to draft. I Will refactor the code.